lib/streamlit/elements/widgets/number_input.py

```diff
@@ -73,47 +75,82 @@ def deserialize(
 
 
 class NumberInputMixin:
+    # For easier readability, all the arguments with un-changing types across these overload signatures have been
+    # collapsed onto a single line.
+
+    # fmt: off
+    # If "min_value: int" is given and all other numerical inputs are "int"s or not provided (value optionally being "min"), return "int"
```

Comment:
nit: while we don't have a linting rule on that yet, we started to keep comments also  < 89 chars